### ISSUE TITLE
azurerm_resource_group_template_deployment - API version information for RP "Microsoft.Compute" ("")was not found - disks

### ISSUE THREAD

**Issue by morbendor** [hashicorp/terraform-provider-azurerm#26348]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.0

### AzureRM Provider Version

3.108.0

### Affected Resource(s)/Data Source(s)

azurerm_resource_group_template_deployment

### Terraform Configuration Files

```hcl
https://drive.google.com/file/d/1F4Q1fhIplXHAhoBw8_Y09l55IF0NoXAu/view?usp=sharing
```

### Debug Output/Panic Output

```shell
│ Error: removing items provisioned by this Template Deployment: API version information for RP "Microsoft.Compute" ("") was not found - nestedResource="/subscriptions/6f-ded2-4f21-b5fe-95276272403c/resourceGroups/rg-networking-test-dev/providers/Microsoft.Compute/disks/vSocket2disk"
```

### Expected Behaviour

The nested Microsoft.Compute/disks resource should have been deleted successfully.

### Actual Behaviour

The provider errors and is not able to remove the nested resource:

` Error: removing items provisioned by this Template Deployment: API version information for RP "Microsoft.Compute" ("") was not found - nestedResource="/subscriptions/a0-ded2-4f21-b5fe-95276272403c/resourceGroups/rg-networking-test-dev/providers/Microsoft.Compute/disks/vSocket2disk"`

### Steps to Reproduce

terraform apply -auto-approve
terraform destroy -auto-approve

**Comment by teowa** [comment]
Hi @morbendor , thanks for submitting the issue!
From the shared configurations, I can find that your template used in `azurerm_resource_group_template_deployment` is a nested template, which deploy a `Microsoft.Resources/deployments` resource, and the `Microsoft.Resources/deployments` resource then deploy a `Microsoft.Compute/disks` resource. This is simiar to the case in https://github.com/hashicorp/terraform-provider-azurerm/issues/25458.

**Comment by rcskosir** [comment]
@teowa Do you think your PR #25466 will also fix this issue?

**Comment by teowa** [comment]
Yes, PR#25466 is intended to fix this, it is still under review.

**Comment by katbyte** [comment]
It appears this issue covers the same ground as #25458 (azurerm_resource_group_template_deployment - API version information for RP "Microsoft.Insights" ("") was not found - activityLogAlerts), which is still open and has more of the discussion, so we are going to close this one out and keep it all in one place.

Please follow and thumbs up #25458 for updates. If these are actually different, please say so in a comment there or open a new issue referencing this one. Thank you!
